net/veth.go

```diff
@@ -272,19 +273,37 @@ func subnets(addrs []netlink.Addr) map[string]struct{} {
 	return subnets
 }
 
+func addNatRule(ipt *iptables.IPTables, rulespec ...string) error {
+	// Loop until we get an exit code other than "temporarily unavailable"
+	for {
+		if err := ipt.AppendUnique("nat", "WEAVE", rulespec...); err != nil {
+			if ierr, ok := err.(*iptables.Error); ok {
+				if status, ok := ierr.ExitError.Sys().(syscall.WaitStatus); ok {
+					// (magic exit code 4 found in iptables source code; undocumented)
+					if status == 4 {
```

Comment:
`status` is actually a bitfield which combines the exit code with a number of flags (see `main 2 wait`):
```
$ cat test.go 
package main

import "fmt"
import "os/exec"
import "syscall"

func main() {
	err := exec.Command("false").Run()
	fmt.Println(err.(*exec.ExitError).Sys().(syscall.WaitStatus))
	fmt.Println(err.(*exec.ExitError).Sys().(syscall.WaitStatus).ExitStatus())
}
$ go run test.go 
256
1
```
You need to do `status.ExitStatus() == 4` instead.